lib/fluent/parser.rb

```diff
@@ -345,6 +352,16 @@ def convert_field_type!(record)
           end
         }
       end
+
+      def convert_value_to_nil(value)
+        if @null_empty_string
+          value = (value == '') ? nil : value
+        end
+        if @null_value_pattern
```

Comment:
`if value and @null_value_pattern` is more clear to tell returning nil when `nil =~ @null_value_pattern`